File: Frontend/hkCmsMobile/app/types/achievement.ts
import { Result } from "./result";

// Defining TS interfaces for the data structures
export interface Achievement {
    id: number;
    name: string;
    url: string;
    description?: string | "";
  }
  
  export interface Progress {
    icon: string;
    message: string;
    buttonText: string;
  }
  
  export interface AchievementCard {
    id: number;
    title: string;
    type: "achievements" | "progress";
    achievements?: Achievement[] | null;
    progress?: Result;
  }
  
  
  export interface UserAchievementDto {
    PlayerId: string;
    AchievementId: number | null;
  }

File: BakOverskriftene.Models/Models/Achievement.cs
namespace BakOverskriftene.Domain.Models
{

    public class Achievement
    {
        
        public int Id { get; set; } //PK
        public string Name { get; set; }
        public string Url { get; set; }
        public string Description { get; set; }

        // Foreign Keys
        public string? PlayerId { get; set; }
        public int? SectionId { get; set; }


    }
}